.github/workflows/proposal-pr-check.yml

```diff
@@ -0,0 +1,43 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+   changed_files:
+    runs-on: ubuntu-latest 
+    name: Test added-OFEP proposals for status
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get added files
+        id: added-files
+        uses: tj-actions/changed-files@v37
+        with: 
+          files: |
+            **.md
+      
+      - if: ${{ steps.added-files.outputs.added_files_count == 0 }}
+        name: No check needed
+        run: |
+          echo "No OFEP proposals were added"
+          exit 0
+        
+      - name: List all added files
+        id: list-added-files
+        if: ${{ steps.added-files.outputs.added_files_count != 0 }}
+        run: |
+          for file in ${{ steps.added-files.outputs.added_files }}; do
+            echo "$file was added"
+            echo "status=$(grep -e "State" "$file" | tr -d "## State: " | tr -d '"')" >> "$GITHUB_OUTPUT"
+          done
+
+      - if: ${{ (steps.added-files.outputs.added_files_count != 0) && (steps.list-added-files.outputs.status != 'REJECTED') && (steps.list-added-files.outputs.status != 'WITHDRAWN') && (steps.list-added-files.outputs.status != 'APPROVED') }}
+        name: Exit with error
+        run: |
+          echo "status is not approved"
+          exit 1
```

Comment:
```suggestion
      - name: Verify statuses
        id: list-added-files
        if: ${{ steps.added-files.outputs.added_files_count > 0 }}
        run: |
          #!/bin/bash

          for file in ${{ steps.added-files.outputs.added_files }}; do
            while IFS= read -r line; do
              if [[ $line == "## State:"* ]]; then
                value=${line#*State:}
                value=${value# } # Remove leading whitespace
                if [[ "$value" != "REJECTED" && "$value" != "WITHDRAWN" && "$value" != "APPROVED" ]]; then
                  echo "Unmatched value detected in file: $file"
                  echo "Detected status: $value"
                  echo "Please set the status to 'REJECTED', 'WITHDRAWN', or 'APPROVED' before merging."
                  exit 1
                fi
              fi
            done < "$file"
          done

          echo "All files checked successfully."
          exit 0
```

Reply:
I didn't understand why this change is required. Could you please explain?

Reply:
The current logic only evaluates the state of the last file in the list of changed files. This change will error as soon as an issue is detected. That ensures all changed files have the proper status.

It's a bit of an edge case since it's unlikely anyone would include multiple OFEPs in a single pr.

Reply:
Yup, I also deliberately ignored that case earlier but now I feel it might be better to include it.